lib/std/builtin/assembly.zig

```diff
@@ -0,0 +1,1026 @@
+        v3: bool = false,
+        v4: bool = false,
+        v5: bool = false,
+        v6: bool = false,
+        v7: bool = false,
+        v8: bool = false,
+        v9: bool = false,
+        v10: bool = false,
+        v11: bool = false,
+        v12: bool = false,
+        v13: bool = false,
+        v14: bool = false,
+        v15: bool = false,
+        v16: bool = false,
+        v17: bool = false,
+        v18: bool = false,
+        v19: bool = false,
+        v20: bool = false,
+        v21: bool = false,
+        v22: bool = false,
+        v23: bool = false,
+        v24: bool = false,
+        v25: bool = false,
+        v26: bool = false,
+        v27: bool = false,
+        v28: bool = false,
+        v29: bool = false,
+        v30: bool = false,
+        v31: bool = false,
+    },
+    .xcore => packed struct {
+        /// Whether the inline assembly code may perform stores to memory
+        /// addresses other than those derived from input pointer provenance.
+        memory: bool = false,
+
+        r0: bool = false,
+        r1: bool = false,
+        r2: bool = false,
+        r3: bool = false,
+        r4: bool = false,
+        r5: bool = false,
+        r6: bool = false,
+        r7: bool = false,
+        r8: bool = false,
+        r9: bool = false,
+        r10: bool = false,
+        r11: bool = false,
+
+        cp: bool = false,
+        dp: bool = false,
+        sp: bool = false,
+        lr: bool = false,
+        sr: bool = false,
+    },
+    .xtensa => packed struct {
+        /// Whether the inline assembly code may perform stores to memory
+        /// addresses other than those derived from input pointer provenance.
+        memory: bool = false,
+
+        sar: bool = false,
+        lbeg: bool = false,
+        lend: bool = false,
+        lcount: bool = false,
+        atomctl: bool = false,
+        scompare1: bool = false,
+        threadptr: bool = false,
+        litbase: bool = false,
+        windowbase: bool = false,
+        windowstart: bool = false,
+        ps: bool = false,
+
+        a0: bool = false,
+        a1: bool = false,
+        a2: bool = false,
+        a3: bool = false,
+        a4: bool = false,
+        a5: bool = false,
+        a6: bool = false,
+        a7: bool = false,
+        a8: bool = false,
+        a9: bool = false,
+        a10: bool = false,
+        a11: bool = false,
+        a12: bool = false,
+        a13: bool = false,
+        a14: bool = false,
+        a15: bool = false,
+
+        b0: bool = false,
+        b1: bool = false,
+        b2: bool = false,
+        b3: bool = false,
+        b4: bool = false,
+        b5: bool = false,
+        b6: bool = false,
+        b7: bool = false,
+        b8: bool = false,
+        b9: bool = false,
+        b10: bool = false,
+        b11: bool = false,
+        b12: bool = false,
+        b13: bool = false,
+        b14: bool = false,
+        b15: bool = false,
+
+        br: bool = false,
```

Comment:
Nit: Suggest moving this before `b0` - `b15` to clarify that it aliases all of them, similar to what you've done for other architectures.